src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentInvokerProvider.cs

```diff
@@ -9,16 +9,16 @@ namespace Microsoft.AspNet.Mvc
     public class DefaultViewComponentInvokerProvider : IViewComponentInvokerProvider
     {
         private readonly IServiceProvider _serviceProvider;
-        private readonly ITypeActivator _typeActivator;
+        private readonly IViewComponentFactory _viewComponentFactory;
         private readonly IViewComponentActivator _viewComponentActivator;
```

Comment:
Please rebase on top of pranav's PR and make the design the same as it is for controllers. If you think this is too big for this PR let's chat.